text-editor/text-editor-003-en.md

```diff
@@ -1 +1,102 @@
 ## Insertion of a tweet : a case study
+
+In this chapter, we will try to give a better insight of how the whole thing is
+working, by breaking down the insertion of a tweet case and the components
+involved in it's workflow.
+
+### Getting the HTML markup from twitter
+
+Keep in mind that our main goal here is to obtain the simplest possible HTML
+representation of our tweet, so it can lately be saved to the API. That way it
+will be easy for us to parse it as a component, and finally display it to the
+end user in the front end application.
+
+The following chart will give you an overview of the actions flow :
+![Insert tweet action flow chart](https://github.com/jaljo/articles/raw/314724209eb27b7b1d61795341934c4783084bee/images/text-editor/RTE.png)
+
+History begins as soon as an user submits the little tweet insertion form.
+Thanks to `redux` action creators, the url of the tweet he wants to insert is
+boxed along with the `INSERT_TWEET` action, which is observed by a first epic.
+
+The HTML markup of a tweet can be obtained by using the twitter API,
+[see here](https://developer.twitter.com/en/docs/twitter-for-websites/embedded-tweets/overview).
+Because of Cross Origin restriction imposed by twitter API, we cannot directly
+call that API from a browser application. To solve this issue, we created a
+dedicated endpoint in our API (which has been mimiced in the `server` folder of
+the POC project).
+
+So the first epic is only responsible to call that endpoint and dispatch a
+`EMBED_TWEET_FETCHED` action that carries out the obtained HTML markup in its
+payload. Its always convenient to isolate API calls in dedicated epics so errors
+can be properly catched and displayed to the end user when something goes wrong.
+Cool thing is, keeping epics as atomic as possible (observe an action, do one
+thing, then dispatch another action) allow them to be easily tested. That is,
+your fellow developpers dont want to break your legs when the prod crashes
+(which will happend, but that's another story).
+
+### Inserting the tweet into the DOM
+
+That last action is observed by another epic which will perform the actual tweet
+insertion into the DOM (as epic are meant to deal with side effects, it's the
+proper place in the project to do that). How is that ?
+
+[See gist here](https://gist.github.com/jaljo/4c8f83acc48766b930d7d584fe9ed22b)
```

Comment:
To improve readability of this file, I think it'd be good to put the actual code in this file too. A link to the gist can be placed below the code ;)